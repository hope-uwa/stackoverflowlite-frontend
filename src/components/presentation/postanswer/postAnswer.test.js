import React from 'react';
import { mount } from 'enzyme';
import PostAnswer from './postAnswer';

describe('display banner', () => {
  const mockFn = jest.fn();
  const setup = () => {
    const props = {
      id: '',
      post: () => {}
    };
    return mount(<PostAnswer {...props} />);
  };
  it('should show page is loading', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('', () => {
    const wrapper = setup();
    expect(wrapper.find('#answerQuestion').text()).toEqual('Add Answer');
  });

  it('should handle input changes successfully', () => {
    const wrapper = setup();
    wrapper.find('#answer').simulate(
      'change',
      {
        target: {
          name: 'body',
          value: 'this is an answer'
        }
      }
    );
    wrapper.find('#answerQuestion').simulate('click', { preventDefault() {} });
    expect(mockFn.mock.calls[0]).toEqual(undefined);
  });

  it('should submit successfully', () => {
    const wrapper = setup();
    wrapper.find('#answerQuestion').simulate('click', { preventDefault() {} });
    expect(mockFn.mock.calls.length).toBe(0);
  });
});
