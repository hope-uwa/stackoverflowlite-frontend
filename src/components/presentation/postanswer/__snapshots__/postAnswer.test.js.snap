// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display banner should show page is loading 1`] = `
<PostAnswer
  id=""
  post={[Function]}
>
  <div
    className="row m12 mls0 pb3"
  >
    <div
      className="answers-card pt1 pb1"
    >
      <div
        className=" smhide answer__avatar"
      />
      <div
        className="answer__box align-right"
      >
        <form
          id="answer-form"
        >
          <h4
            className="align-left"
          >
            Your Answer
          </h4>
          <textarea
            className="answer-text"
            id="answer"
            name="body"
            onChange={[Function]}
          />
          <button
            className="btn btn-success btn-sm"
            id="answerQuestion"
            onClick={[Function]}
          >
            Add Answer
          </button>
        </form>
      </div>
      <div
        className="answer__response black-text"
      />
    </div>
  </div>
</PostAnswer>
`;
